File: 3. Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        current_substring = ''
        longest_substring = ''

        for i in range(len(s)):
            if s[i] not in current_substring:
                current_substring += s[i]
            else:
                current_substring = current_substring[current_substring.index(s[i]) + 1:] + s[i]
            if len(current_substring) > len(longest_substring):
                longest_substring = current_substring
            
        return len(longest_substring)

File: 3. Longest Substring Without Repeating Characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let currentSubstring = ''
    let longestSubstring = ''

    for (let i = 0; i < s.length; i++) {
        if(!currentSubstring.includes(s[i])) {
            currentSubstring += s[i]
        }
        else {
            currentSubstring = currentSubstring.slice(currentSubstring.indexOf(s[i]) + 1) + s[i]
        }
        if(currentSubstring.length > longestSubstring.length) {
            longestSubstring = currentSubstring
        }
    }

    return(longestSubstring.length)
};
